hedera-mirror-monitor/src/main/java/com/hedera/hashgraph/sdk/ManagedNode.java

```diff
@@ -0,0 +1,342 @@
+
+    protected ManagedNode(ManagedNodeAddress address, ExecutorService executor) {
+        this.executor = executor;
+        this.address = address;
+        this.currentBackoff = Client.DEFAULT_MIN_BACKOFF;
+        this.minBackoff = Client.DEFAULT_MIN_BACKOFF;
+        this.maxBackoff = Client.DEFAULT_MAX_BACKOFF;
+    }
+
+    protected ManagedNode(N node, ManagedNodeAddress address) {
+        this.address = address;
+
+        this.executor = node.executor;
+        this.minBackoff = node.minBackoff;
+        this.maxBackoff = node.maxBackoff;
+        this.backoffUntil = node.backoffUntil;
+        this.currentBackoff = node.currentBackoff;
+        this.attempts = node.attempts;
+        this.lastUsed = node.lastUsed;
+        this.useCount = node.useCount;
+    }
+
+    /**
+     * Create an insecure version of this node
+     *
+     * @return
+     */
+    abstract N toInsecure();
+
+    /**
+     * Create a secure version of this node
+     * @return
+     */
+    abstract N toSecure();
+
+    abstract KeyT getKey();
+
+    /**
+     * Get the address of this node
+     *
+     * @return
+     */
+    ManagedNodeAddress getAddress() {
+        return address;
+    }
+
+    /**
+     * Get the minimum backoff
+     * @return
+     */
+    Duration getMinBackoff() {
+        return minBackoff;
+    }
+
+    /**
+     * Set the minimum backoff
+     *
+     * @param minBackoff
+     * @return
+     */
+    N setMinBackoff(Duration minBackoff) {
+        this.minBackoff = minBackoff;
+
+        // noinspection unchecked
+        return (N) this;
+    }
+
+    /**
+     * Get the maximum backoff
+     * @return
+     */
+    Duration getMaxBackoff() {
+        return maxBackoff;
+    }
+
+    /**
+     * Set the minimum backoff
+     *
+     * @param maxBackoff
+     * @return
+     */
+    N setMaxBackoff(Duration maxBackoff) {
+        this.maxBackoff = ManagedNode.this.maxBackoff;
+
+        // noinspection unchecked
+        return (N) this;
+    }
+
+    /**
+     * Get the number of times this node has received a bad gRPC status
+     * @return
+     */
+    long getAttempts() {
+        return attempts;
+    }
+
+    /**
+     * Determines if this is node is healthy.
+     * Healthy means the node has either not received any bad gRPC statuses, or if it has received bad gRPC status then
+     * the node backed off for a period of time.
+     *
+     * @return
+     */
+    boolean isHealthy() {
+        return backoffUntil < System.currentTimeMillis();
+    }
+
+    /**
+     * Used when a node has received a bad gRPC status
+     */
+    synchronized void increaseDelay() {
+        this.attempts++;
+        this.backoffUntil = System.currentTimeMillis() + this.currentBackoff.toMillis();
+        this.currentBackoff = Duration.ofMillis(Math.min(this.currentBackoff.toMillis() * 2, this.maxBackoff.toMillis()));
+    }
+
+    /**
+     * Used when a node has not received a bad gRPC status.
```

Comment:
nit: Try wording in the affirmative instead of double negative
```suggestion
     * Used when a node has received a good gRPC status after receiving multiple bad gRPC status.
```
Can then remove the '(receiving several bad gRPC status)' since the context is provided earlier

Reply:
As discussed offline, this is a copied SDK file so trying to minimize changes to just fix the bug.

Reply:
Seems to be SDK copied file so not worth the effort resolving this here.